src/cascadia/TerminalApp/IDirectKeyListener.idl

```diff
@@ -8,7 +8,7 @@ namespace TerminalApp
     // If you update this one, please update the one in TerminalControl\TermControl.idl
     // If you change this interface, please update the guid.
     // If you press F7 or Alt and get a runtime error, go make sure both copies are the same.
-    [uuid("339e1a87-5315-4da6-96f0-565549b6472b")] interface IDirectKeyListener {
-        Boolean OnDirectKeyEvent(UInt32 vkey, Boolean down);
+    [uuid("0ddf4edc-3fda-4dee-97ca-a417ee3dd510")] interface IDirectKeyListener {

```

Comment:
why bump the uuid?

Reply:
you're supposed to do this on every change to an existing interface, technically. The interface is really supposed to unique the vtable layout that allows for a conformant class to be called by an unknown caller -- the chief feature of COM :smile:

Also because it says it in the rules on line 9

Reply:
It's not applicable here but it's good to get in the habit. If somebody had taken a dependency on this interface solely by layout and old UUID, new consumers are no longer compatible with it. If we didn't change the uuid, consumers of the new layout would _report as compatible_ but get the parameter layout wrong (unpacking the first 8 bits of the Boolean as the scancode (!!))

Reply:
that's what I thought, but I'm glad to have this typed out for posterity 😄